WordPress/gutenberg
# [Mobile] Add react-native-fast-image

**Related PRs:**
- `Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/4989
- `WordPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/16832
- `WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/18995

## What?
This PR adds the [React Native FastImage](https://github.com/DylanVann/react-native-fast-image) library that improves the performance of image rendering in the editor.

**Note**: There are other changes needed to improve the performance even more with the following PR https://github.com/WordPress/gutenberg/pull/42178.

## Why?
The goal is to improve the performance of the editor when there are images (Gallery, and Image blocks). The React Native FastImage library uses [Glide (Android)](https://github.com/bumptech/glide) and [SDWebImage (iOS)](https://github.com/rs/SDWebImage) so it helps with rendering and memory usage.

## How?
By replacing the built-in Image from React Native and using FastImage instead:

- BlockStyles component
- Image component: used in the Image and Gallery block (v1 and v2)

## Testing Instructions

**Note**: Test with **both** the demo app and the WordPress apps using the builds from the PRs linked above.

### Test case 1 - Image block

- Open the app
- Add an Image block
- Select an image to upload
- Expect the image to be uploaded and rendered correctly

### Test case 2 - Gallery block

- Open the app
- Add a Gallery block
- Select a few images to upload
- Expect the images to be uploaded and rendered correctly

## Screenshots or screencast 

### Test case 1 - Image block

https://user-images.githubusercontent.com/4885740/177776476-ef60fae6-a0ac-49f4-b244-05cd60b2c4d9.mov

### Test case 2 - Gallery block

https://user-images.githubusercontent.com/4885740/177775840-5ec70b68-b0dc-4c0f-a47e-4c973b7f64dd.mov
